File: 双指针/python/leftRightPointer.py
class LeftRightPointer:

    @staticmethod
    def two_num_sum(nums, target):
        """
        给定一个升序排列的数组nums, 找到数组中两个数使得它们的和等于target
        """
        left = 0
        right = len(nums) - 1
        while left < right:
            if nums[left] + nums[right] == target:
                return [left, right]
            elif nums[left] + nums[right] > target:
                right -= 1
            elif nums[left] + nums[right] < target:
                left += 1
        return [-1, -1]


if __name__ == "__main__":
    nums = [2, 7, 11, 15]
    target = 10
    print(LeftRightPointer.two_num_sum(nums, target))

File: DoublePoint/cpp/leftRightPointer.cpp
#include <iostream>
#include <vector>

using namespace std;

bool leftRightPointer(const vector<int>& nums, int target)
{
    vector<int>::const_iterator left = nums.begin();
    vector<int>::const_iterator right = nums.end() - 1;
    while(left != right)
    {
        if(*left + *right == target)
            return true;
        else if (*left + *right < target)
            left++;
        else
            right--;
    }
    return false;
}

int main()
{
    vector<int> nums = {2, 7, 11, 15};
    cout << leftRightPointer(nums, 9) << endl;
    cout << leftRightPointer(nums, 10) << endl;
    return 0;
}